### ISSUE TITLE
Invalid tokens cause random "Connection reset by peer", which prevents from finishing a sending batch

### ISSUE THREAD

**Issue by albertboada** [immobiliare/ApnsPHP#34]
Been testing the library for quite a while and, although it's definitely the best one out there for PHP, I'm almost getting insane with some random behaviour I've noticed mixing valid and invalid tokens.

After writing all push notifications to the socket (both valid and invalid tokens mixed up), we just wait through `select_stream()` for any error feedback. When the error is available (which it is, since `$nChangedStreams` is `1`), we call `_updateQueue()`, which calls `_readErrorMessage()`, which does a `fread()` to the socket.

Although `$nChangedStreams` is indeed always `1` whenever it has to be, `_readErrorMessage`'s `fread()` fails **once in a while**, throwing away the following warning (both in UNIX and Windows environments):

```
Warning: fread(): SSL: Connection reset by peer in path_to_project/ApnsPHP/Push.php on line 302 
```

Something is getting in the way between the `select_stream()` check, and the following `fread()`, but I'm not sure anymore which end is to blame here.

As mentioned, this only happens sometimes, not always, not never. When it doesn't, everything goes well and all valid notifications end up being sent, with its proper retries, as expected. On the other hand, when it fails, the queue is emptied, since the library considers there's no error if `fread()` returns nothing, and so retries are lost and execution ends (having lost the remaining pending notifications).

Am I the only one to notice this behaviour? Or is it a common issue? Advice is appreciated.

Thanks for your time.

**Comment by junghyun-kim** [comment]
Any updates on this issue?

**Comment by lethalman** [comment]
I'm not able to reproduce this issue using the code in https://github.com/immobiliare/ApnsPHP/issues/34#issuecomment-20441441 . I correctly see 20 messages being set, and 20 errors at the end. I don't see any connection reset by peer error in the middle.

**Comment by junghyun-kim** [comment]
Frankly speaking, I'm using another language - ruby. And I am facing the same problem. 'Connection reset by peer' after sending messages including some invalid tokens. But it is not always. I tested with 1000 messages with 10~20% of invalid tokens. And sometimes I can see ECONNRESET (connection reset by peer). 

I'm sorry to ask here. I think this issue is not a problem of this library. I found same problem at node-apn. (https://github.com/argon/node-apn/issues/137)
